src/regressor.rs

```diff
@@ -245,14 +254,32 @@ impl Regressor {
                 )
                 .unwrap();
 
-                if layernorm == NNLayerNorm::BeforeRelu {
+                if layernorm == NNLayerNorm::BeforeActivation {
                     output =
                         block_normalize::new_normalize_layer_block(&mut bg, &mi, output).unwrap();
                 }
-                if activation == NNActivation::Relu {
-                    output = block_relu::new_relu_block(&mut bg, &mi, output).unwrap();
+
+                match activation {
+                    NNActivation::None => {}
+                    NNActivation::Relu => {
+                        output = block_relu::new_relu_block(&mut bg, &mi, output).unwrap();
+                        println!("Relu layer");
+                    }
+                    NNActivation::LeakyRelu => {
+                        output = block_leaky_relu::new_leaky_relu_block(&mut bg, output).unwrap();
+                        println!("LeakyRelu layer");
+                    }
+                    NNActivation::Tanh => {
+                        output = block_tanh::new_tanh_block(&mut bg, output).unwrap();
+                        println!("Tanh layer");
+                    }
+                    NNActivation::Sigmoid => {
+                        output = block_sigmoid::new_sigmoid_block(&mut bg, output).unwrap();
+                        println!("Sigmoid layer");
+                    }
                 }
-                if layernorm == NNLayerNorm::AfterRelu {
+
+                if layernorm == NNLayerNorm::AfterActivation {
```

Comment:
Nice work! Btw, worth seeing speed comparison with just relu in this case, probably not critical but would be interesting to see

Reply:
sure I will test if for a regular sequential training scenario